### PR TITLE
Failed L2 swaps don't display Failed transaction activity state

### PR DESCRIPTION
**Pull request by derHowie**
Fixes [TEAM2-237]
Transaction.status doesn't exist on the transaction confirmation object anymore. We now invoke txConfirmationObject.wait() to inspect the transaction receipt to determine the status.
https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse

Screen recordings / screenshots
https://recordit.co/WMGFxVEjH6

## What to test
https://github.com/rainbow-me/rainbow/tree/%40derHowie/check-tx-receipts-test-branch
The branch above cuts `gasLimit` in half on swaps and enabled the transaction watcher so that mainnet swaps will fail and we will see the changes in the pending transaction logic execute.

* ensure failed transactions appear correctly in the tx list on the branch above
* ensure successful transactions behave normally on the branch that isn't broken (@derHowie/check-tx-receipts)

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/redux/data.ts [hunk 1]
```diff
@@ -1597,8 +1597,28 @@ export const dataWatchPendingTransactions = (
...
+            const title = getTitle({
+              protocol: tx.protocol,
+              status: updatedPending.status,
+              type: tx.type,
+            });
+            updatedPending.title = title;
+            updatedPending.pending = false;
+            updatedPending.minedAt = minedAt;
+          }
+          if (txObj && !isZero(receipt?.status || 0)) {
```
Comment (estebanmino) [inline src/redux/data.ts:1621]: maybe 
```
if (receipt && txObj && !isZero(receipt?.status || 0))
```
?

File: src/redux/data.ts [hunk 2]
```diff
@@ -1597,8 +1597,28 @@ export const dataWatchPendingTransactions = (
...
+            }
+            const title = getTitle({
+              protocol: tx.protocol,
+              status: updatedPending.status,
+              type: tx.type,
+            });
+            updatedPending.title = title;
+            updatedPending.pending = false;
+            updatedPending.minedAt = minedAt;
+          }
```
Comment (estebanmino) [inline src/redux/data.ts:1620]: if i'm following this well, line 1641 would be changing the `updatedPending` obj anyways?

Comment (derHowie) [reply src/redux/data.ts:1620]: > if i'm following this well, line 1641 would be changing the `updatedPending` obj anyways?

Yeah, good call. Looks like we only need to assign a value to `updatedPending.status` here.

**Review by estebanmino — APPROVED** [review]
tried several txs, polygon swaps and mainnet txs 
image

mainnet txs were speeded up, all looking solid 🪨

File: src/redux/data.ts [hunk 3]
```diff
@@ -1597,8 +1597,23 @@ export const dataWatchPendingTransactions = (
           }
           const minedAt = Math.floor(Date.now() / 1000);
           txStatusesDidChange = true;
-          // @ts-expect-error `txObj` is not typed as having a `status` field.
-          if (txObj && !isZero(txObj.status)) {
+          let receipt;
```
Comment (jinchung) [inline src/redux/data.ts:1600]: hey QQ: any reason we cannot use `p.getTransactionReceipt(txHash)` directly on line 1588?

**Pull request merged**